packages/flutter/test/painting/text_span_test.dart

```diff
@@ -476,4 +477,156 @@ void main() {
+        const <(int, TextStyle)>[(0, TextStyle(fontSize: 1000)), (1, TextStyle(fontSize: 100)), (5, TextStyle(fontSize: 10))],
+      );
+      expect(
+        toStyleRuns(simpleSpan.updateAttributes(const InlineSpanAttributes(fontSize: 1000), const TextRange(start: 1, end: 2))),
+        const <(int, TextStyle)>[(0, TextStyle(fontSize: 100)), (1, TextStyle(fontSize: 1000)), (2, TextStyle(fontSize: 100)), (5, TextStyle(fontSize: 10))],
+      );
+      expect(
+        toStyleRuns(simpleSpan.updateAttributes(const InlineSpanAttributes(fontSize: 1000), const TextRange(start: 2, end: 5))),
+        const <(int, TextStyle)>[(0, TextStyle(fontSize: 100)), (2, TextStyle(fontSize: 1000)), (5, TextStyle(fontSize: 10))],
+      );
+      expect(
+        toStyleRuns(simpleSpan.updateAttributes(const InlineSpanAttributes(fontSize: 1000), const TextRange(start: 0, end: 5))),
+        const <(int, TextStyle)>[(0, TextStyle(fontSize: 1000)), (5, TextStyle(fontSize: 10))],
+      );
+      expect(
+        toStyleRuns(simpleSpan.updateAttributes(const InlineSpanAttributes(fontSize: 1000), const TextRange(start: 5, end: 6))),
+        const <(int, TextStyle)>[(0, TextStyle(fontSize: 100)), (5, TextStyle(fontSize: 1000)), (6, TextStyle(fontSize: 10))],
+      );
+      expect(
+        toStyleRuns(simpleSpan.updateAttributes(const InlineSpanAttributes(fontSize: 1000), const TextRange(start: 6, end: 6))),
+        const <(int, TextStyle)>[(0, TextStyle(fontSize: 100)), (5, TextStyle(fontSize: 10))],
+      );
+      expect(
+        toStyleRuns(simpleSpan.updateAttributes(const InlineSpanAttributes(fontSize: 1000), const TextRange(start: 6, end: 7))),
+        const <(int, TextStyle)>[(0, TextStyle(fontSize: 100)), (5, TextStyle(fontSize: 10)), (6, TextStyle(fontSize: 1000)), (7, TextStyle(fontSize: 10))],
+      );
+      expect(
+        toStyleRuns(simpleSpan.updateAttributes(const InlineSpanAttributes(fontSize: 1000), const TextRange(start: 6, end: 10))),
+        const <(int, TextStyle)>[(0, TextStyle(fontSize: 100)), (5, TextStyle(fontSize: 10)), (6, TextStyle(fontSize: 1000))],
+      );
+    });
+
+    test('can add and delete recognizers', () {
+      final GestureRecognizer recognizer = TapGestureRecognizer();
+      addTearDown(recognizer.dispose);
+      final int length = simpleSpan.toPlainText().length;
+      final TextSpan withRecognizer = simpleSpan.updateAttributes(
+        InlineSpanAttributes(recognizer: Left(recognizer)),
+        const TextRange(start: 1, end: 9),
+      );
+      for (int i = 0; i < length; i++) {
+        final TextSpan span = withRecognizer.getSpanForPosition(TextPosition(offset: i))! as TextSpan;
+        expect(span.recognizer, 1 <= i && i < 9 ? recognizer : null, reason: 'index = $i');
+      }
+
+      final TextSpan partiallyDeleted = withRecognizer.updateAttributes(
+        const InlineSpanAttributes(recognizer: InlineSpanAttributes.remove),
+        const TextRange(start: 5, end: 6),
+      );
```

Comment:
The remove API is kinda awkward. Could we split up updateAttributes into addAttribute and removeAttribute to make this more straight-forward?

Reply:
The downside of that is we'll need two separate methods and two separate attribute sets (one for update and one for removal) for basically the same thing on `InlineSpan`. Maybe introducing sentinel values for removal is the way?